elements/lisk-chain/src/block_reward.ts

```diff
@@ -158,13 +158,18 @@ export const applyFeeAndRewards = async (
 	// This is necessary only for genesis block case, where total fee is 0, which is invalid
 	// Also, genesis block cannot be reverted
 	generator.balance += givenFee > 0 ? givenFee : BigInt(0);
-	const totalFeeBurntStr = await stateStore.chain.get(CHAIN_STATE_BURNT_FEE);
-	let totalFeeBurnt = BigInt(totalFeeBurntStr ?? 0);
+	const totalFeeBurntBuffer = await stateStore.chain.get(CHAIN_STATE_BURNT_FEE);
+	let totalFeeBurnt = BigInt(
+		totalFeeBurntBuffer ? JSON.parse(totalFeeBurntBuffer.toString('utf8')) : 0,
+	);
 	totalFeeBurnt += givenFee > 0 ? totalMinFee : BigInt(0);
 
 	// Update state store
 	stateStore.account.set(generatorAddress, generator);
-	stateStore.chain.set(CHAIN_STATE_BURNT_FEE, totalFeeBurnt.toString());
+	stateStore.chain.set(
+		CHAIN_STATE_BURNT_FEE,
+		Buffer.from(JSON.stringify(totalFeeBurnt.toString()), 'utf8'),
+	);
```

Comment:
Can we use some utility function to convert back and forth from the buffer, rather repeating the same code.

Reply:
This needs to be updated using the lisk-codec